arcgis-ios-sdk-samples/Content Display Logic/Controllers/AppInfoViewController.swift

```diff
@@ -14,40 +14,167 @@
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if Section.allCases[section] == .poweredBy {
+            return UITableView.automaticDimension
+        } else {
+            return .zero
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section.allCases[section] {
+        case .appInfo:
+            return appInfos.count
+        case .links:
+            return links.count
+        case .poweredBy:
+            return 1
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: Section.allCases[indexPath.section].cellIdentifier, for: indexPath)
+        switch Section.allCases[indexPath.section] {
+        case .appInfo:
+            cell.textLabel?.text = appInfos[indexPath.row].title
+            cell.detailTextLabel?.text = appInfos[indexPath.row].detail
+        case .links:
+            cell.textLabel?.text = links[indexPath.row].title
+        case .poweredBy:
+            cell.textLabel?.text = "ArcGIS Runtime SDK for iOS"
+        }
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        switch Section.allCases[indexPath.section] {
+        case .links:
+            UIApplication.shared.open(links[indexPath.row].url)
+        case .poweredBy:
+            UIApplication.shared.open(URLs.developers)
+        default:
+            return
+        }
+    }
+}
+
+// MARK: - Enums and Structs
+
+private extension AppInfoViewController {
+    struct AppInfo {
+        let title: String
+        let detail: String
+    }
+    
+    struct Link {
+        let title: String
+        let url: URL
+    }
+    
+    /// A convenience type for the table view sections.
+    enum Section: Int, CaseIterable {
+        case appInfo, links, poweredBy
+        
+        var titleForHeader: String {
+            switch self {
+            case .appInfo:
+                return "App Info"
+            case .links:
+                return "Useful Links"
+            case .poweredBy:
+                return "Powered By"
+            }
+        }
+        
+        var cellIdentifier: String {
+            switch self {
+            case .appInfo:
+                return "RightDetailCell"
+            case .links:
+                return "LinkCell"
+            case .poweredBy:
+                return "LinkCell"
+            }
+        }
     }
     
-    @IBAction func runtimeSDKForiOSAction() {
-        UIApplication.shared.open(.developers)
+    enum Strings {
+        /// The manually assigned marketing version, e.g. "100.13.0.1".
+        static let appMarketingVersionNumber = "100.13.0.1"
```

Comment:
Unfortunately no, because this 4-component version string is not allowed in Apple's semantic versioning rules. This is a user facing version, rather than the app bundle version.

We have to update this string manually, together with the GitHub release tag, and the App Store Connect Marketing Version each time we release. The decision was captured in `common-samples/issues/2996`